### PR TITLE
Allow joint model group to have use IK solvers with multiple tip frames

### PR DESCRIPTION
**Pull request by davetcoleman**
**Note:** _This commit will not build until the pending PR for moveit_core is merged ros-planning/moveit_core#149_

Adds functionality to support multiple tip-frames (non-chain kinematics) in IK solutions. 
- For `JointModelGroup`, changes `canSetStateFromIK` to check all available tip frames
- In `RobotState`, changes `setFromIK` to send a pose for all necessary tip frames that an IK solver requires. For instance, if your tree-kinematics IK solver supports 4 tips (4 end effectors, such as two hands and two feet), it will send an IK request to `KinematicsBase` with 4 poses.

### CODE REVIEW

File: robot_model/src/joint_model_group.cpp [hunk 1]
```diff
@@ -554,29 +554,49 @@ bool moveit::core::JointModelGroup::canSetStateFromIK(const std::string &tip) co
 {
   const kinematics::KinematicsBaseConstPtr& solver = getSolverInstance();
   if (!solver || tip.empty())
+  {
+    logError("Cannot set state, no solver or tip is empty");
```
Comment (hersh) [inline robot_model/src/joint_model_group.cpp]: These logError() printouts need to be removed.  The point of the function is for the caller to know if setFromIK() is going to work, so the caller is doing the right thing, even if the return value is false.

File: robot_model/src/joint_model_group.cpp [hunk 2]
```diff
@@ -554,29 +554,49 @@ bool moveit::core::JointModelGroup::canSetStateFromIK(const std::string &tip) co
...
     return false;
-  const std::string &tip_frame = solver->getTipFrame();
-  if (tip_frame.empty())
+  }
+
+  const std::vector<std::string> &tip_frames = solver->getTipFrames();
+
+  if (tip_frames.empty())
+  {
+    logDebug("Group %s has no tip frame(s)", name_.c_str());
```
Comment (hersh) [inline robot_model/src/joint_model_group.cpp:563]: This logError() needs to be removed also.

File: robot_state/src/robot_state.cpp [hunk 3]
```diff
@@ -1332,9 +1389,8 @@ bool moveit::core::RobotState::setFromIK(const JointModelGroup *jmg, const Eigen
     // compute the IK solution
     std::vector<double> ik_sol;
     moveit_msgs::MoveItErrorCodes error;
-    if (ik_callback_fn ?
-        solver->searchPositionIK(ik_query, seed, timeout, consistency_limits, ik_sol, ik_callback_fn, error, options) :
-        solver->searchPositionIK(ik_query, seed, timeout, consistency_limits, ik_sol, error, options))
+    // \todo no function call that accepts multiple poses and no callback has been created. is it really needed?
```
Comment (hersh) [inline robot_state/src/robot_state.cpp]: This "todo" can be done (and then removed from this pull request please) by changing the default implementation of KinematicsBase::searchPositionIK() which takes multiple tip frames to check if the callback function is NULL and if so, call the searchPositionIK() without a callback argument.  See https://github.com/ros-planning/moveit_core/pull/149/files#r10730403.

Comment (davetcoleman) [inline robot_state/src/robot_state.cpp]: Fixed, thanks.

File: robot_model/src/joint_model_group.cpp (hunk 2 above)
Comment (davetcoleman) [inline robot_model/src/joint_model_group.cpp:563]: You mean `logDebug()`? I think its useful feedback to aid the user in understanding why their planning group is unable to use an IK solver. This area is currently kind of difficult IMHO to debug.

File: robot_model/src/joint_model_group.cpp (hunk 1 above)
Comment (davetcoleman) [inline robot_model/src/joint_model_group.cpp]: Removed.

File: robot_model/src/joint_model_group.cpp (hunk 2 above)
Comment (hersh) [inline robot_model/src/joint_model_group.cpp:563]: My mistake, logDebug() is fine to leave in.

**Pull request merged**